### ISSUE TITLE
Add event context for hotkeys subsystem

### ISSUE THREAD

**Issue by neithernut** [waysome/waysome#423]
The events emitted by the hotkeys subsystem don't carry a context with them. Such a context should include a reference to the surface under the cursor. It should (most naturally) have a method for querying that surface.
Blocks #420.

**Comment by matthiasbeyer** [comment]
There is a `sturct ws_hotkey_event` type, should an PR for this issue extend this type?

**Comment by neithernut** [comment]
Currently, the event _contains_ the context, but there _are_ situations where the context differs.
We _may_ use the `ws_hotkey_event` as the root of a hirarchy of context, but i'm not sure if that's what we want.

**Comment by matthiasbeyer** [comment]
I don't understand why the context can differ. How is that possible, can you elaborate or provide an example why this could happen?

**Comment by neithernut** [comment]
If the hotkeys subsystem emits an event, it should contain date like the surface under the mouse (as described above). If you, however, want to trigger execution from a plugin via an event, you _may_ want to pass some "arguments" or other data to the transaction. Finally, a shutdown may also trigger an event. In this case, you definitely do not care about the surface under the cursor.
TL;DR: There are different sources of events and the intent behind events differs, too. Having _one_ context doesn't make sense.

**Comment by matthiasbeyer** [comment]
Couldn't this be generated by using `ws_value_set`, which can contain several objects whenever there are  several values in the context?

**Comment by neithernut** [comment]
This _is_ an option, but not a nice one.

**Comment by neithernut** [comment]
Mh.. we just realized that there's a better option.
